clients/src/main/java/org/apache/kafka/common/network/NetworkReceive.java

```diff
@@ -89,10 +126,17 @@ public long readFromReadableChannel(ReadableByteChannel channel) throws IOExcept
                     throw new InvalidReceiveException("Invalid receive (size = " + receiveSize + ")");
                 if (maxSize != UNLIMITED && receiveSize > maxSize)
                     throw new InvalidReceiveException("Invalid receive (size = " + receiveSize + " larger than " + maxSize + ")");
-
-                this.buffer = ByteBuffer.allocate(receiveSize);
+                requestedBufferSize = receiveSize; //may be 0 (?!)
```

Comment:
I originally had a <= 0 check which triggered while testing my code. it surprised me (as evident by the comment) but I decided to live with it instead of tracking it down.